independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Beans.java

```diff
@@ -520,9 +521,21 @@ static BeanInfo resolveAmbiguity(Collection<BeanInfo> resolved) {
         }
         if (resolvedAmbiguity.size() == 1) {
             return resolvedAmbiguity.get(0);
+        } else if (resolvedAmbiguity.size() == 0) {
+            // all beans were default beans, we attempt to sort them based on priority
+            resolvedAmbiguity = new ArrayList<>(resolved);
+            resolvedAmbiguity.sort(Beans::compareDefaultBeans);
+            Integer highest = getDefaultBeanPriority(resolvedAmbiguity.get(0));
+            for (Iterator<BeanInfo> iterator = resolvedAmbiguity.iterator(); iterator.hasNext();) {
+                if (!highest.equals(getDefaultBeanPriority(iterator.next()))) {
+                    iterator.remove();
+                }
+            }
+            if (resolvedAmbiguity.size() == 1) {
+                selected = resolvedAmbiguity.get(0);
+            }
```

Comment:
We shouldn't really fall through here, like, ever? Why should we apply the ambiguity resolution algorithm again, if there's multiple default beans with the same highest priority?

Reply:
Good point, there should be a `return selected;` statement below this.
There is no need to continue the algorithm.

Reply:
Should be fixed now, thanks